### ISSUE TITLE
Unable to resolve module `./$.html` (bad module resolution in bundler)

### ISSUE THREAD

**Issue by haggholm** [facebook/react-native#20721]
- [x] Review the documentation: https://facebook.github.io/react-native
  - [x] Search for existing issues: https://github.com/facebook/react-native/issues
  - [x] Use the latest React Native release: https://github.com/facebook/react-native/releases

## Environment

Note: Although it looks like a custom React Native build, it is in fact vanilla 0.56 except with a bugfix backported (https://github.com/facebook/react-native/pull/18522). The bugfix pertains to Java code and this bug is in the JS bundling.

```
  React Native Environment Info:
    System:
      OS: Linux 4.15 Ubuntu 18.04.1 LTS (Bionic Beaver)
      CPU: x64 Intel(R) Core(TM) i7-2600K CPU @ 3.40GHz
      Memory: 4.94 GB / 31.39 GB
      Shell: 4.4.19 - /bin/bash
    Binaries:
      Node: 8.11.1 - /usr/bin/node
      Yarn: 1.5.1 - /usr/local/bin/yarn
      npm: 6.3.0 - /usr/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      Android SDK:
        Build Tools: 23.0.1, 25.0.0, 25.0.2, 26.0.0, 26.0.1, 26.0.2, 26.0.3, 27.0.1, 27.0.3
        API Levels: 23, 24, 26, 27
    npmPackages:
      @client/react: ^1.0.13 => 1.0.68 
      @client/react-native: ^1.0.13 => 1.0.68 
      react-native: github:haggholm/react-native#asyncstorage-fix-0.56 => 0.56.0 
```
[skip envinfo]

## Description

In an attempt to reduce the number of imports and speed up app launch, I decided to try to use `@babel/preset-env` with `useBuiltins: 'usage'`. However, this fails when trying to resolve `core-js` modules:

```
Unable to resolve module `./$.html` from `/home/petter/owl/client-react-native-navigation/root/node_modules/react-native/node_modules/core-js/modules/$.task.js`: The module `./$.html` could not be found from `/home/petter/owl/client-react-native-navigation/root/node_modules/react-native/node_modules/core-js/modules/$.task.js`. Indeed, none of these files exist:

  * `$.html`
  * `/home/petter/owl/client-react-native-navigation/root/node_modules/react-native/node_modules/core-js/modules/$.html/index(.native||.android.js|.native.js|.js|.android.json|.native.json|.json)`
```
In other words, faced with a request to load `$.html`, it seems the RN bundler assumes that either

1. It will have that exact name, or
2. It will be a directory where we can look for an `index` entry.

What it does *not* try to resolve is the correct option, which is `$.html.js`.

**Comment by haggholm** [comment]
> Can you reproduce the issue on the latest release, v0.56?

I *did* reproduce it on the latest release, v0.56. Of course, I don’t expect you to understand the caveat, since you’re just a bot, but let’s be clear on the record…

**Comment by haggholm** [comment]
Digging a little bit (ongoing), I find that the problem is that by the time it tries to load the file, it has already classified it as an asset, due presumably to the `.html` extension:

```json
  "candidates": {
    "file": {
      "type": "asset",
      "name": "$.html"
    },
    "dir": {
      "type": "sourceFile",
      "filePathPrefix": "/mnt/scratch/shared/petter/witkit-devenv/client-react-native-navigation/root/node_modules/react-native/node_modules/core-js/modules/$.html/index",
      "candidateExts": [
        ".native",
        "",
        ".android.js",
        ".native.js",
        ".js",
        ".android.json",
        ".native.json",
        ".json"
      ]
    }
  }
```

So it looks for the non-existent asset file `$.html` and therefore never checks for the source file `$.html.js`.

**Comment by haggholm** [comment]
The issue has not to my knowledge been fixed, just ignored.

**Comment by Slapbox** [comment]
Can anyone comment on this issue?

**Comment by nyl9488** [comment]
fixed??

**Comment by cpojer** [comment]
This seems like an issue with Metro, not React Native. Note that you should be using `babel-preset-env` with React Native, there is no added benefit as there is only one, known, environment.
